src/main/java/edu/harvard/iq/dataverse/externaltools/ExternalToolServiceBean.java

```diff
@@ -45,45 +45,50 @@
     }
 
     /**
-     * @param type
+     * @param type explore or configure
      * @return A list of tools or an empty list.
      */
-    public List<ExternalTool> findByType(Type type) {
-        return findByType(type, null);
+    public List<ExternalTool> findDatasetToolsByType(Type type) {
+        String nullContentType = null;
+        return findByScopeTypeAndContentType(ExternalTool.Scope.DATASET, type, nullContentType);
     }
 
     /**
-     * @param type
-     * @param contentType - mimetype
+     * @param type explore or configure
      * @return A list of tools or an empty list.
      */
-    public List<ExternalTool> findByType(Type type, String contentType) {
-
-        List<ExternalTool> externalTools = new ArrayList<>();
+    public List<ExternalTool> findFileToolsByType(Type type) {
+        String nullContentType = null;
+        return findByScopeTypeAndContentType(ExternalTool.Scope.FILE, type, nullContentType);
+    }
 
-        //If contentType==null, get all tools of the given ExternalTool.Type 
-        TypedQuery<ExternalTool> typedQuery = contentType != null ? em.createQuery("SELECT OBJECT(o) FROM ExternalTool AS o WHERE o.type = :type AND o.contentType = :contentType", ExternalTool.class)
-                : em.createQuery("SELECT OBJECT(o) FROM ExternalTool AS o WHERE o.type = :type", ExternalTool.class);
-        typedQuery.setParameter("type", type);
-        if (contentType != null) {
-            typedQuery.setParameter("contentType", contentType);
-        }
-        List<ExternalTool> toolsFromQuery = typedQuery.getResultList();
-        if (toolsFromQuery != null) {
-            externalTools = toolsFromQuery;
-        }
-        return externalTools;
+    /**
+     * @param type explore or configure
+     * @param contentType file content type (MIME type)
+     * @return A list of tools or an empty list.
+     */
+    public List<ExternalTool> findFileToolsByScopeAndContentType(Type type, String contentType) {
```

Comment:
this method should by not be "by scope" since scope is file or type (personally I think it's overkill to have separate methods for file and dataset, but at least they wrappers around a single method that takes in scope as a parameter)

Reply:
Whoops! Fixed in 94f812724 by removing "scope". Good catch!